src/ops/validate.rs

```diff
@@ -162,11 +162,10 @@ impl ValidateOp for super::Conditional {
         for (i, (child, optype)) in children.into_iter().enumerate() {
             let OpType::Case(case_op) = optype else {panic!("Child check should have already checked valid ops.")};
             let sig = &case_op.signature;
-            let predicate_value = &self.predicate_inputs[i];
-            if sig.input[0..predicate_value.len()] != predicate_value[..]
-                || sig.input[predicate_value.len()..] != self.other_inputs[..]
-                || sig.output != self.outputs
-            {
+            if let Some(expected_inputs) = self.case_input_row(i) {
+                if sig.input == expected_inputs && sig.output == self.outputs {
+                    continue;
```

Comment:
I had to do it this way (i.e. inverting the predicate) to use `let Some` as I couldn't find a way to compare an `Option<&TypeRow>` with an `Option<TypeRow>`. Seems like I needed the opposite of `as_ref`. Maybe you can see a way?

Reply:
(that is, I wanted to write something like `if Some(sig.input) != &self.case_input_row(i)` but with some `&`s or similar on the LHS but couldn't work out what)